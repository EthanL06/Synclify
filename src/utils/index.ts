import type { State } from "../types/state"

export const hasVideos = () => {
  return document.getElementsByTagName("video").length > 0
}

export const setState = (
  tabId: number,
  roomId: string,
  state?: State,
  videoFound?: boolean
) => {
  return Object.assign(state ?? {}, {
    [tabId]: {
      roomId: roomId,
      videoFound: videoFound ?? state?.[tabId]?.videoFound ?? false
    }
  })
}
