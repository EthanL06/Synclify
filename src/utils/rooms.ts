type RoomsList = { [tabId: number]: string }

export const storeRoom = (rooms: string, r: RoomsList) => {
  if (rooms === undefined) return JSON.stringify(r)
  const obj = JSON.parse(rooms)
  console.log("after parse", obj)
  Object.assign(obj, r)
  console.log("after assignement", obj)
  console.log("JSON string", JSON.stringify(obj))
  return JSON.stringify(obj)
}

export const deleteRoom = (rooms: string, tabid: number) => {
  const obj: RoomsList = JSON.parse(rooms)
  if (obj[tabid]) delete obj[tabid]
  return JSON.stringify(obj)
}

export const parseRooms = (rooms: string): RoomsList => {
  if (rooms === undefined) return {}
  return JSON.parse(rooms)
}

export {}
