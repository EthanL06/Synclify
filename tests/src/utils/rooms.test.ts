/* eslint-disable quotes */
/* eslint-disable prefer-const */
/* eslint-disable @typescript-eslint/no-explicit-any */
import * as rooms from "../../../src/utils/rooms"
// @ponicode
describe("rooms.parseRooms", () => {
  test("0", () => {
    let result: any = rooms.parseRooms(undefined)
    expect(result).toMatchSnapshot()
  })

  test("1", () => {
    let result: any = rooms.parseRooms('{ "1000959510": "EB875" }')
    expect(result).toMatchSnapshot()
  })

  test("2", () => {
    let result: any = rooms.parseRooms("")
    expect(result).toMatchSnapshot()
  })
})

// @ponicode
describe("rooms.deleteRoom", () => {
  test("0", () => {
    let result: any = rooms.deleteRoom('{ "1000959510": "EB875" }', 1)
    expect(result).toMatchSnapshot()
  })

  test("1", () => {
    let result: any = rooms.deleteRoom(undefined, -5.48)
    expect(result).toMatchSnapshot()
  })

  test("2", () => {
    let result: any = rooms.deleteRoom(undefined, 1)
    expect(result).toMatchSnapshot()
  })

  test("3", () => {
    let result: any = rooms.deleteRoom('{ "1000959510": "EB875" }', 1000959510)
    expect(result).toMatchSnapshot()
  })

  test("4", () => {
    let result: any = rooms.deleteRoom(undefined, 100)
    expect(result).toMatchSnapshot()
  })

  test("5", () => {
    let result: any = rooms.deleteRoom(undefined, -Infinity)
    expect(result).toMatchSnapshot()
  })
})

// @ponicode
describe("rooms.storeRoom", () => {
  test("0", () => {
    let result: any = rooms.storeRoom(undefined, { 123: "ABCD" })
    expect(result).toMatchSnapshot()
  })

  test("1", () => {
    let result: any = rooms.storeRoom('{ "1000959510": "EB875" }', {
      123: "ABCD",
      456: "DEFG"
    })
    expect(result).toMatchSnapshot()
  })

  test("3", () => {
    let result: any = rooms.storeRoom(undefined, {})
    expect(result).toMatchSnapshot()
  })

  test("4", () => {
    let result: any = rooms.storeRoom(undefined, { 123: "ABCD", 456: "DEFG" })
    expect(result).toMatchSnapshot()
  })

  test("5", () => {
    let result: any = rooms.storeRoom("", { 123: "" })
    expect(result).toMatchSnapshot()
  })
})
