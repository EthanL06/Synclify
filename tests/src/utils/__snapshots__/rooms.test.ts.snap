// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rooms.deleteRoom 0 1`] = `"{"1000959510":"EB875"}"`;

exports[`rooms.deleteRoom 1 1`] = `undefined`;

exports[`rooms.deleteRoom 2 1`] = `undefined`;

exports[`rooms.deleteRoom 3 1`] = `"{}"`;

exports[`rooms.deleteRoom 4 1`] = `undefined`;

exports[`rooms.deleteRoom 5 1`] = `undefined`;

exports[`rooms.parseRooms 0 1`] = `null`;

exports[`rooms.parseRooms 1 1`] = `
{
  "1000959510": "EB875",
}
`;

exports[`rooms.parseRooms 2 1`] = `null`;

exports[`rooms.storeRoom 0 1`] = `"{"123":"ABCD"}"`;

exports[`rooms.storeRoom 1 1`] = `"{"123":"ABCD","456":"DEFG","1000959510":"EB875"}"`;

exports[`rooms.storeRoom 3 1`] = `"{}"`;

exports[`rooms.storeRoom 4 1`] = `"{"123":"ABCD","456":"DEFG"}"`;

exports[`rooms.storeRoom 5 1`] = `"{"123":""}"`;
